checks/raw/shell_download_validate.go

```diff
@@ -397,11 +399,7 @@ func collectExecuteFiles(startLine, endLine uint, node syntax.Node, cmd, pathfn
 
 // Npm install docs are here.
 // https://docs.npmjs.com/cli/v7/commands/npm-install
-func isNpmUnpinnedDownload(cmd []string) bool {
-	if len(cmd) == 0 {
```

Comment:
without the length check, this code can panic when accessing `cmd[0]`

Reply:
I moved this validation to a step before.
In `collectUnpinnedPakageManagerDownload` we evaluate if the command is a go, pip, pip, npm, choco or nuget download. In this function I added this validation to do an early return if the command is empty. Otherwise, it will not be empty when it reaches `isNpmDownload` and similar isXYXDownload functions. Since isXYXDownload functions are only used in `collectUnpinnedPakageManagerDownload`, I believe the code will not panic.
I can readd this validation to the isXYXDownload methods for precaution if you'd like.